### PR TITLE
[RFR] Hack had invalid logic and invalid bz_blocker

### PR DESCRIPTION
**Pull request by psav**
The logic was always returning False for the BZ, and should have been NOT'ing the blocks

{{pytest: cfme/tests/infrastructure/test_provider_discovery.py -k test_discover_infra}}

### CODE REVIEW

**Comment by psav** [comment]
This needs to be validated for 5.9 especially

File: cfme/common/provider.py [hunk 1]
```diff
@@ -618,7 +618,8 @@ def clear_providers(cls):
         from cfme.utils.appliance import current_appliance as app
         app.rest_api.collections.providers.reload()
         # cfme 5.9 doesn't allow to remove provider thru api
-        if app.version < '5.9' or (app.version >= '5.9' and BZ(1501941).blocks):
+        bz_blocked = BZ(1501941, forced_streams=['5.9']).blocks
+        if app.version < '5.9' or (app.version >= '5.9' and not bz_blocked):
```
Comment (izapolsk) [inline cfme/common/provider.py]: yeah, my bad. I slightly updated if statement after testing code.

**Comment by izapolsk** [comment]
@psav, providers aren't removed in 5.8 and 5.7. Maybe provider.all issue ?

**Comment by psav** [comment]
This is a much better piece of code now :)

**Comment by izapolsk** [comment]
@psav, I agree it looks better. However, I added that logic with grouping provider types in order to get rid of navmazing delays.

**Pull request merged**